compiler/rustc_mir/src/interpret/machine.rs

```diff
@@ -176,8 +176,10 @@ pub trait Machine<'mir, 'tcx>: Sized {
     ) -> InterpResult<'tcx>;
 
     /// Called to evaluate `Abort` MIR terminator.
-    fn abort(_ecx: &mut InterpCx<'mir, 'tcx, Self>) -> InterpResult<'tcx, !> {
-        throw_unsup_format!("aborting execution is not supported")
+    fn abort(_ecx: &mut InterpCx<'mir, 'tcx, Self>, msg: String) -> InterpResult<'tcx, !> {
+        use crate::const_eval::ConstEvalErrKind;
+
+        Err(ConstEvalErrKind::Abort(msg).into())
```

Comment:
This should not be the default implementation. Instead please leave the default implementation as-is, and only implement the new `abort` for the CTFE machine defined here:

https://github.com/rust-lang/rust/blob/a68864b68859e5848865bd392ce6d892b16f5cdd/compiler/rustc_mir/src/const_eval/machine.rs#L245